flink-streaming-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/Kafka09Fetcher.java

```diff
@@ -285,7 +293,14 @@ public void commitSpecificOffsetsToKafka(Map<KafkaTopicPartition, Long> offsets)
 
 		if (this.consumer != null) {
 			synchronized (consumerLock) {
-				this.consumer.commitSync(offsetsToCommit);
+				if (!commitInProgress) {
+					commitInProgress = true;
+					this.consumer.commitAsync(offsetsToCommit, offsetCommitCallback);
+				}
+				else {
+					LOG.warn("Committing previous checkpoint's offsets to Kafka not completed. " +
```

Comment:
Possibly yes. But on the other hand, this should be pretty visible if it happens.
I would expect that with proper options to participate in group checkpoint committing, most Flink jobs run without committing to Kafka/ZooKeeper.